src/Neo.IO/ByteArrayComparer.cs

```diff
@@ -20,27 +20,32 @@ internal class ByteArrayComparer : IComparer<byte[]>
         public static readonly ByteArrayComparer Default = new(1);
         public static readonly ByteArrayComparer Reverse = new(-1);
 
-        private readonly int direction;
+        private readonly int _direction;
 
         private ByteArrayComparer(int direction)
         {
-            this.direction = direction;
+            _direction = direction;
         }
 
-        public int Compare(byte[] x, byte[] y)
+        public int Compare(byte[]? x, byte[]? y)
         {
-            return direction > 0
-                ? CompareInternal(x, y)
-                : -CompareInternal(x, y);
+            if (ReferenceEquals(x, y)) return 0;
+            if (x is null && y is not null) return -y.Length;
+            if (y is null && x is not null) return x.Length;
```

Comment:
If x is empty, it will return 0 that means is the same, I think that it's better to return -1 and +1 only when null

Reply:
Now tell if i'm wrong; i sometimes have trouble with stuff like this.

Currently flow goes
1. if `x` is same instance as `y`
2. if `x` is `null` then return `x` is less than `y` length.
3. if `y` is `null` then return `x` is greater than `y` length
4. if `x` and `y` is not `null` then return the comparison
5. _the only thing left is:_ if `x` and `y` is `null` return `0` aka the same.